### DISCUSSION TITLE
Question about upgrading the Haydes Canyon Nuc from the original guide

### DISCUSSION THREAD

**Discussion by Jbmac123** [osy86/HaC-Mini#1108] [Q&A]
Hello everyone,
I followed the original guide for the Haydes Canyon Nuc : [https://osy.gitbook.io/hac-mini-guide](url) and it has worked great for the last few years. My knowledge of how to do all of this is pretty much non existent, I used the guide word for word like following a recipe and I was surprised that even with this it worked as well as it did 😂.

I would like to upgrade my Haydes Canyon Nuc to the latest macOS that it is able to run, is there a clear and easy to follow step by step guide like how the original was written? or can someone talk me through the steps please?

Thank you so much for your question, and I apologize in advance for being a complete noob in regards to all of this.

**Comment by Medizinmann-MD** [comment]
Well I try to write a short guide here - but this might not be 100% noob-/foolproof.

You need an USB stick with at least 16 GB (I would opt for something fast – that helps speed up things – like i.e. Samsung Fit or Bar Plus 64 GB) and I recommend a second fresh SSD (best one compatible with MacOS like i.e. WD SN570, SN770 or SN850(X)).

Get yourself the latest Version of OCLP and make an installer USB for Sonoma.
https://dortania.github.io/OpenCore-Legacy-Patcher/

If you use a Broadcom WiFi (i.e. BCM94360NG like me) get yourself the EFI from the SashonIT here (I can report this works mostly* fine for me):
https://github.com/osy/HaC-Mini/discussions/1095#discussioncomment-8756736

If you use the Intel WiFi you could look into the version from Deftaudio (I hadn’t time to test this myself though):
https://github.com/osy/HaC-Mini/discussions/1095#discussioncomment-9347116

Now get yourself OCAT:
https://github.com/5T33Z0/OC-Little-Translated/tree/main/D_Updating_OpenCore

…and ProperTree:
https://github.com/corpnewt/ProperTree

Mount the ESP with OCAT and choose the installer stick, create a directory “EFI” and copy “Boot” and “OC” from one of the zip-archives above to it.
Now load the config.plist(found in oc) from this EFI and create a new serial etc. using OCAT (for Mac mini 8,1) and save the resulting EFI – you might need to upgrade and sync OpenCore to 0.9.9 (as a time of writing choose latest) beforehand (find it in the OCAT menu bar under Edit) to prevent error/corruption of the config.plist.
Copy the Platforminfo from this EFI to the other two EFIs on the stick using ProperTree – or you could copy your own old one later if you like to preserve it.

Now rename the “config.plist” to something else and the “config Update Kopie.plist” to config.plist.
Or deactivate the IOSkywalker kexts manually.

Copy OCAT and OCLP to the installer USB stick.
Boot the installer USB and install Sonoma on the fresh SSD.
Run OCLP – don’t reboot now, use OCAT to mount ESP on the installer USB and the SSD and copy the EFI from the USB to the SSD.
Rename the “config.plist” back to “config update.plist” or whatever you like and rename the former config.plist again to config.plist – be sure that all the IOSkywalker kexts are now activated.

Reboot and the system should be up and running.
Now you can either migrate you old setup from a time machine backup or directly from the old SSD using the migration assistant in MacOS Sonoma.

And voila done.

Before you login with your AppleID you need to decide if you like use the new serial – than this is a new machine on your account (shouldn’t be a problem) or if you like to preserve the old serial and copy this one to this machine by coping the Platforminfo into your current config.plist using ProperTree.

Before the next Update rename again the config.plist accordingly – run the update, run OCLP and rename config.plist back again and reboot.

I hope this helps.

*For me everything works so far - only sometimes Bluetooth doesn’t work after wakeup from screen off – but simply turning it off and on again fixes this glitch – but it works most of the time so this is IMHO a minor glitch.

**Comment by Jbmac123** [comment]
Thank you so much for taking the time to make that amazing explanation, I'm
seriously blown away by your kindness! I have a few projects on Premiere
Pro that I need to get finished before risking the upgrade but as soon as
they are done I will give it a go and let you know how it all went.
Thank you again.

**Reply by Medizinmann-MD** [reply]
> Thank you so much for taking the time to make that amazing explanation, I'm seriously blown away by your kindness! 

You are welcome!

> I have a few projects on Premiere Pro that I need to get finished before risking the upgrade but as soon as they are done I will give it a go and let you know how it all went. Thank you again.

That is why I recommend the risk free route with a fresh/new/2nd SSD - this way you can always go back to your known good install (!) - and don't forget you need a compatible SSD model like i.e. WD SN 570, 770 or 850(X)!
